src/plcc/parse_spec/parse_syntactic_spec/parse_syntactic_spec.py

```diff
@@ -0,0 +1,133 @@
+from typing import List
+from plcc.load_spec.load_rough_spec.parse_dividers import Divider
+from plcc.load_spec.load_rough_spec.parse_lines import Line
+import re
+from re import Match
+from .parse_syntactic_structs import (
+    SyntacticSpec,
+    SyntacticRule,
+    Symbol,
+    CapturingTerminal,
+    RepeatingSyntacticRule,
+    StandardSyntacticRule,
+    LhsNonTerminal,
+    RhsNonTerminal,
+    Terminal,
+    CapturingSymbol,
+    MalformedLHSError,
+    MalformedBNFError,
+)
+
+
+def parse_left(line: Line, lhs: str) -> LhsNonTerminal:
+    match = re.match(r"<(?P<non_terminal>\S*)>(?::(?P<alt_name>\S+))?\s*", lhs)
+    if match is None:
+        raise MalformedLHSError(line)
+
+    non_terminal = match["non_terminal"]
+    alt_name = match["alt_name"]
+    return LhsNonTerminal(non_terminal, alt_name)
+
+
+def parse_right(line: Line, rhs: str) -> List[Symbol]:
+    total = []
+    if not rhs:
+        return total
+    symbols = rhs.split(" ")
+    for symbol in symbols:
+        if symbol == "":
+            continue
+        if symbol.startswith("#") or symbol is None:
+            return total
+        total.append(parse_symbol(symbol))
+    return total
+
+
+def parse_symbol(symbol: str) -> Symbol:
+    capturing = re.match(r"<(?P<name>\S*)>(?P<alt_name>\S+)?", symbol)
+
+    if capturing:
+        name = capturing["name"]
+        alt_name = capturing["alt_name"]
+        return parse_capturing(name, alt_name)
+    return Terminal(symbol)
+
+
+def parse_capturing(name: str, alt_name: str) -> CapturingSymbol:
+    terminal = re.match(r"[A-Z_]+", name)
+
+    if alt_name is not None:
+        alt_name = alt_name.strip(":")
+
+    if terminal:
+        return CapturingTerminal(name, alt_name)
+    return RhsNonTerminal(name, alt_name)
+
+
+def is_skippable(line: Line) -> bool:
+    empty = re.match(r"^\s*$", line.string)
+    return empty is not None or line.string.startswith("#")
+
+
+def build_matches(
+    line: str,
+) -> tuple[Match[str] | None, Match[str] | None, Match[str] | None]:
+
+    standard = re.match(r"^\s*(?P<lhs><\S+>(?::\S+)?)\s*::=(?:\s(?P<rhs>.*))?", line)
+    separated = re.match(
+        r"^\s*(?P<lhs><\S+>(?::\S+)?)\s*\*\*=\s(?P<rhs>.*)(?P<separator>\+.*)",
+        line,
+    )
+    repeating = re.match(r"^\s*(?P<lhs><\S+>(?::\S+)?)\s*\*\*=\s(?P<rhs>.*)", line)
+
+    return standard, separated, repeating
```

Comment:
Keeping the function definition on one line would improve readability in this case, even if it is longer than typical, and the separated variable can also be kept as one line, as it would make the code shorter and simpler to read, also the comma after the parameter type can be removed as there is only one parameter.